docs/src/pages/[platform]/theming/dark-mode/react.mdx

```diff
@@ -38,15 +38,64 @@ Amplify UI comes with a default dark mode that you can use.
 
 _Note: the Amplify UI theme and any overrides like dark mode are scoped to the ThemeProvider. Changing the color mode in the example will only affect the example code._
 
-<Example>
-  <DefaultDarkMode />
+<Tabs>
+  <TabItem title="JavaScript">
+    <Example>
+      <DefaultDarkMode />
 
-  <ExampleCode>
-    ```tsx file=./examples/DefaultDarkMode.tsx
-    ```
-    
-  </ExampleCode>
-</Example>
+      <ExampleCode>
+```jsx
+import * as React from 'react';
+import {
+  defaultDarkModeOverride,
+  ThemeProvider,
+  Card,
+  Text,
+  ToggleButton,
+  ToggleButtonGroup,
+} from '@aws-amplify/ui-react';
+
+export const DefaultDarkMode = () => {
+  const [colorMode, setColorMode] = React.useState('system');
+  const theme = {
+    name: 'my-theme',
+    overrides: [defaultDarkModeOverride],
+  };
+
+  return (
+    <ThemeProvider theme={theme} colorMode={colorMode}>
+      <Card>
+        <ToggleButtonGroup
+          value={colorMode}
+          isExclusive
+          onChange={(value) => setColorMode(value)}
+        >
+          <ToggleButton value="light">Light</ToggleButton>
+          <ToggleButton value="dark">Dark</ToggleButton>
+          <ToggleButton value="system">System</ToggleButton>
+        </ToggleButtonGroup>
+        <Text>Current color mode: {colorMode}</Text>
+      </Card>
+    </ThemeProvider>
+  );
+};
+```
+        
+      </ExampleCode>
+    </Example>
+  </TabItem>
+  <TabItem title="TypeScript">
+    <Example>
+      <DefaultDarkMode />
+
+      <ExampleCode>
+        ```tsx file=./examples/DefaultDarkMode.tsx
```

Comment:
Wondering if we should move the example code in _DefaultDarkMode.tsx_ into this file as well to make it things more clear down the road. Just thinking that a future reader may be confused why there is a separate file for the `tsx` example while the `jsx` sample is included in this file. What do you think?

Reply:
I agree with example codes being confusing.

We are however actually rendering the JSX component from `DefaultDarkMode.tsx` -- that is the `<DefaultDarkMode />` right above this line, so I don't think we want to hard code the `tsx` here as well.

Reply:
This is why `.jsx` file made more sense in terms of consistency; but I was worried about it confusing the `next` bundler.

Reply:
Maybe we could postfix the extension with `.jsx.snippet`, just to signal that this file does not intend to be get rendered?

Reply:
cc @dbanksdesign, do you have a suggestion?

Reply:
I definitely want to keep the pattern (at least for React right now) of rendering the component right next to the code that gets shown, like this:

```
      <DefaultDarkMode />

      <ExampleCode>
        ```tsx file=./examples/DefaultDarkMode.tsx
```

To ensure the code we are showing is valid and accurate. Obviously that is not practical for all cases like some of our CSS ones and this one where it's a bit ridiculous to have JSX and TSX files for on minor difference. I agree with the comment above that for now rendering the TSX file and hardcoding the JSX in the MDX file is the most practical until we come up with a better long-term approach for JS/TS examples.